src/main/scala/essential_scala/Ex3Classes.scala

```diff
@@ -74,14 +74,23 @@ object Ex3Classes extends App {
     colour: String,
     food: String
   )
+
+  // 3.3.2.1 Friendly Person Factory
+  println(Person("James Bond"))
+  println(Person("Chingachgook The Great Snake"))
+  println(Person("Chingachgook"))
+  println(Person(""))
 }
 
 // 3.3.2.1 Friendly Person Factory
-class Person(firstName: String, lastName: String)
+class Person(firstName: String, lastName: String) {
+  override def toString: String = s"$firstName $lastName"
+}
 object Person {
   def apply(name: String): Person = {
-    val parts = name.split(" ")
-    new Person(parts(0), parts(1))
+    name.split(" ").toList match {
+      case f :: l => new Person(f, l.mkString(" "))
```

Comment:
If the contract of `.split()` will change in future, and will start to return empty sequence for the empty string, the compiler will not recognize the error here because you are using partial function, and partial function is defined as `PartialFunction[-A, +B] extends (A) => B`

Reply:
I fixed the function by adding the missed case. I hadn't been seen it because of using a scala console instead of an sbt shell in IDEA. It allows to faster start a neccessary file with Ctrl Shift F10, but all compile warnings and errors leaves in the sbt shell.